frontend/src/components/pages/Dashboard/components/DropoffCard.tsx

```diff
@@ -1,107 +1,172 @@
-import { CheckIcon, CloseIcon, RepeatIcon } from "@chakra-ui/icons";
-import { Box, HStack, Stack, Text } from "@chakra-ui/react";
-import { format, isToday, isTomorrow } from "date-fns";
+import { Box, Stack, Text, VStack } from "@chakra-ui/react";
+import { format, parse } from "date-fns";
 import React from "react";
 import { useHistory } from "react-router-dom";
 
+import DonorAPIClient from "../../../../APIClients/DonorAPIClient";
+import VolunteerAPIClient from "../../../../APIClients/VolunteerAPIClient";
 import { getFrequencyColor } from "../../../../constants/DaysInWeek";
 import * as Routes from "../../../../constants/Routes";
 import { Schedule } from "../../../../types/SchedulingTypes";
+import {
+  dateHeadingText,
+  getAssistanceType,
+} from "../../../../utils/DashboardUtils";
 import { DonationFrequency } from "../../Scheduling/types";
 
-const DropoffCard = ({ schedule }: { schedule: Schedule }): JSX.Element => {
+const DropoffCardSubInformation = ({
+  description,
+  value,
+  isFrequencyBlock,
+  frequency,
+  frequencyColorScheme,
+}: {
+  description: string;
+  value: string;
+  isFrequencyBlock?: boolean;
+  frequency?: string;
+  frequencyColorScheme?: string;
+}) => (
+  <VStack spacing="4px" alignItems="left" minWidth="10vw">
+    <Text textTransform="uppercase" color="hubbard.100" textStyle="mobileSmall">
+      {description}
+    </Text>
+    <Text color="black.100" textStyle="desktopBody">
+      {isFrequencyBlock && (
+        <Box as="span" textStyle="mobileBodyBold" color={frequencyColorScheme}>
+          {frequency === DonationFrequency.ONE_TIME ? "One time" : frequency}
+        </Box>
+      )}
+      {value}
+    </Text>
+  </VStack>
+);
+
+const DropoffCard = ({
+  schedule,
+  isDonorView,
+  isPublicView = false,
+}: {
+  schedule: Schedule;
+  isDonorView: boolean;
+  isPublicView?: boolean;
+}): JSX.Element => {
   const {
     startTime,
     endTime,
     id,
     frequency,
     volunteerNeeded,
-    recurringDonationEndDate,
+    isPickup,
+    volunteerTime,
+    volunteerId,
+    donorId,
+    categories,
   } = schedule;
+  const history = useHistory();
   const frequencyColorScheme = getFrequencyColor(frequency);
+  const [volunteerAssigned, setVolunteerAssigned] = React.useState("");
+  const [donorName, setDonorName] = React.useState("");
   const startDateLocal = new Date(startTime);
   const startTimeLocal = format(new Date(startTime), "h:mm aa");
   const endTimeLocal = format(new Date(endTime), "h:mm aa");
-  const formattedRecurringEndDate = format(
-    new Date(recurringDonationEndDate),
-    "MMM d, yyyy",
-  );
 
-  const dateHeadingText = (startDate: Date) => {
-    if (isToday(startDate)) {
-      return "Today";
-    }
-    if (isTomorrow(startDate)) {
-      return "Tomorrow";
-    }
-    return format(startDate, "MMM d, yyyy");
-  };
+  React.useEffect(() => {
+    const getVolunteerName = async (vId: string) => {
+      if (volunteerId) {
```

Comment:
should this be vId or volunteerId? I dont even think we need to pass in the id since we have donorId and volunteerId, why introduce a new variable `vId` and `dId`